eng/tools/GenerateFiles/Directory.Build.targets.in

```diff
@@ -2,43 +2,92 @@
-      />
+      Condition="@(KnownAppHostPack->Count()) != '0' AND
+        !(@(KnownAppHostPack->AnyHaveMetadataValue('TargetFramework', '${KnownAppHostPackOrFrameworkReferenceTfm}')))" />
     <KnownFrameworkReference
       Include="@(KnownFrameworkReference->WithMetadataValue('TargetFramework', 'netcoreapp5.0'))"
       TargetFramework="${KnownAppHostPackOrFrameworkReferenceTfm}"
-      Condition="@(KnownFrameworkReference->Count()) != '0' AND !(@(KnownFrameworkReference->AnyHaveMetadataValue('TargetFramework', '${KnownAppHostPackOrFrameworkReferenceTfm}')))"
-      />
+      Condition="@(KnownFrameworkReference->Count()) != '0' AND
+        !(@(KnownFrameworkReference->AnyHaveMetadataValue('TargetFramework', '${KnownAppHostPackOrFrameworkReferenceTfm}')))" />
   </ItemGroup>
 
   <ItemGroup>
-    <!-- Use the same NETCore shared framework as repo built against. -->
+    <!-- Use the same NETCore shared framework as repo built against except when building product code in servicing. -->
     <KnownFrameworkReference
-      Update="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.NETCore.App')->WithMetadataValue('TargetFramework', '${KnownAppHostPackOrFrameworkReferenceTfm}'))"
-      DefaultRuntimeFrameworkVersion="${MicrosoftNETCoreAppRuntimeVersion}"
-      LatestRuntimeFrameworkVersion="${MicrosoftNETCoreAppRuntimeVersion}"
-      TargetingPackVersion="${MicrosoftNETCoreAppRefVersion}" />
+        Update="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.NETCore.App')->WithMetadataValue('TargetFramework', '${KnownAppHostPackOrFrameworkReferenceTfm}'))"
+        LatestRuntimeFrameworkVersion="${MicrosoftNETCoreAppRuntimeVersion}">
+      <!--
+        Change the default shared framework and targeting pack version only when _not_ servicing. Avoid bumping
+        version used in most projects. When servicing, projects can use $(TargetLatestRuntimePatch) to explicitly
+        control whether assemblies build against default (false) or latest (true). When that property is not set, SDK
+        uses default metadata in most cases but published apps e.g. tool projects (again, property not set) use latest.
+
+        !temporary! Also check $(TargetLatestRuntimePatch) here because these metadata changes otherwise increase the
+        minimum versions, making $(TargetLatestRuntimePatch) irrelevant. This helps to avoid problems with current
+        `[assembly: AssemblyVersion(...)]` changes in dotnet/runtime assemblies and our MSBuild tasks.
+      -->
+      <DefaultRuntimeFrameworkVersion Condition=" '$(IsServicingBuild)' != 'true' AND
+          '$(TargetLatestDotNetRuntime)' != 'false' ">${MicrosoftNETCoreAppRuntimeVersion}</DefaultRuntimeFrameworkVersion>
+      <TargetingPackVersion Condition=" '$(IsServicingBuild)' != 'true' AND
+          '$(TargetLatestDotNetRuntime)' != 'false' ">${MicrosoftNETCoreAppRefVersion}</TargetingPackVersion>
```

Comment:
@jkotalik the `AND '$(TargetLatestDotNetRuntime)' != 'false'` additions here are what #27845 needs (in root Directory.Build.targets) until this gets in.

Reply:
Do I need to these changes specifically in this file as well? Sounds like no.